ERCS/erc-7682.md

```diff
@@ -72,10 +72,104 @@ type AuxiliaryFundsCapability = {
+
+```typescript
+// Example wallet_sendCalls type extension
+type WalletSendCallsRequest = {
+  version: string;
+  from: `0x${string}`;
+  chainId: `0x{string}`;
+  atomicRequired: boolean;
+  calls: Array<{
+    to: `0x${string}`;
+    data: string;
+    // ... other call fields ...
+  }>;
+  capabilities?: {
+    auxiliaryFunds?: {
+      assetsRequiredForExecution?: {
+        [asset: `0x${string}`]: {
+          amount: `0x${string}`; // Amount required, as a hex string representing the integer value in the asset's smallest unit
+        };
+      };
+    };
+  };
+};
+```
+
+#### Example Scenario: Cross-Chain Deposit to Aave
+
+Suppose a user wants to deposit DAI into Aave on Chain X, but their wallet address on Chain X has no DAI or native gas token. However, the wallet has access to funds on Chain Y (e.g., Ethereum mainnet). The app, upon detecting the `auxiliaryFunds` capability, can construct a `wallet_sendCalls` request as follows:
+
+```json
+{
+  "calls": [
+    {
+      "to": "0xAaveDAIDepositContractOnChainX",
+      "data": "0xd0e30db0" // Example encoded deposit function call
+    }
+  ],
+  "capabilities": {
+    "auxiliaryFunds": {
+      "assetsRequiredForExecution": {
+        "0x6B175474E89094C44Da98b954EedeAC495271d0F": { // DAI address on Chain X
+          "amount": "0x0de0b6b3a7640000" // 1 DAI
+        }
+      }
+    }
+  }
+}
+```
+
+The wallet, upon receiving this request, can:
+- Bridge or swap the required DAI from Chain Y to Chain X
+- Ensure the user has enough native asset (e.g., ETH) for gas on Chain X
+- Complete the deposit call to Aave on Chain X
+
+This mechanism allows advanced wallets to abstract away the complexity of cross-chain or offchain funding, enabling seamless user experiences even when the user's onchain balance is insufficient on the target chain.
+
+### Auxiliary Funds and Atomic Execution
+
+Auxiliary funds provisioning (bridging, swapping, or other asset retrieval operations) is independent of the `wallet_sendCalls` execution lifecycle. The `atomicRequired` field applies only to the call bundle execution, not to auxiliary funds provisioning.
+
+### Error Codes
+
+The following error codes are defined for auxiliary funds provisioning failures.
+
+| Code     | Message                           | Description |
+|----------|-----------------------------------|-------------|
+| -32080   | Auxiliary funds provisioning failed | Wallet attempted to provision funds but failed (e.g., bridge/swap error). |
+| -32081   | Asset not supported             | The requested asset is not available through the wallet’s auxiliary fund system. |
+| -32082   | Auxiliary funds timeout         | The provisioning operation took too long or timed out. |
```

Comment:
does the dapp need to differentiate between -32080 and -32082?

Reply:
no probably not! Cleaned up here: https://github.com/ethereum/ERCs/pull/1126/commits/dd3a05e0af065da03d7ce5a2819dce45c53d429f